### ISSUE TITLE
Cannot delete the Repositories sync job permanently

### ISSUE THREAD

**Issue by DesistDaydream** [kubeapps/kubeapps#2223]
### Description:
When I create a Repository and delete it, it still triggers sync job.

But the related secrets has been deleted, so the pod cannot be created successfully.And you can still see the deleted repo in the Dashboard
![image](https://user-images.githubusercontent.com/39847025/103087455-850d3600-4622-11eb-8640-de1761ec7f4d.png)

```shell
root@lichenhao:~# kubectl get pod -n kubeapps
NAME                                                              READY   STATUS                       RESTARTS   AGE
apprepo-default-sync-registory-bj-net-1608809400-q6976            0/1     CreateContainerConfigError   0          6m4s
apprepo-kubeapps-sync-bitnami-1608808200-lgfpb                    0/1     Completed                    0          26m
apprepo-kubeapps-sync-bitnami-1608808800-mjclv                    0/1     Completed                    0          16m
apprepo-kubeapps-sync-bitnami-1608809400-nlclm                    0/1     Completed                    0          6m4s
apprepo-kubeapps-sync-registory-bj-net-1608808200-lmcg9           0/1     Completed                    0          26m
apprepo-kubeapps-sync-registory-bj-net-1608808800-v7vsg           0/1     Completed                    0          16m
apprepo-kubeapps-sync-registory-bj-net-1608809400-4v9c6           0/1     Completed                    0          6m4s
apprepo-test-sync-registory-bj-net-1608808200-d84gx               0/1     Completed                    0          26m
apprepo-test-sync-registory-bj-net-1608808800-sfjmw               0/1     Completed                    0          16m
apprepo-test-sync-registory-bj-net-1608809400-2tfwc               0/1     Completed                    0          6m4s
apprepo-test-sync-registory-bj-net-2-1608809400-7xv9n             0/1     CreateContainerConfigError   0          6m4s

root@lichenhao:~# kubectl describe pod -n kubeapps apprepo-test-sync-registory-bj-net-2-1608810600-m6fcb
......
Events:
  Type     Reason     Age                     From               Message
  ----     ------     ----                    ----               -------
  Normal   Scheduled  9m52s                   default-scheduler  Successfully assigned kubeapps/apprepo-test-sync-registory-bj-net-2-1608810600-m6fcb to master-3.bj-test
  Normal   Pulled     7m59s (x12 over 9m51s)  kubelet            Container image "docker.io/bitnami/kubeapps-asset-syncer:2.0.1-scratch-r0" already present on machine
  Warning  Failed     4m49s (x26 over 9m51s)  kubelet            Error: secret "test-apprepo-registory-bj-net-2" not found
```

### Steps to reproduce the issue:
1. create repo 
2. delete repo

### Describe the results you received:
The repo cannot be completely deleted

### Describe the results you expected:
The repo should be completely deleted and no longer trigger the job

### Version of Helm, Kubeapps and Kubernetes:

- Output of `helm version`:

```
root@lichenhao:~# helm version
version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}
```

- Output of `helm list -n <kubeapps-namespace> <kubeapps-release-name>`:

```
root@lichenhao:~# helm ls -n kubeapps
NAME            	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART         	APP VERSION
kubeapps-bj-test	kubeapps 	1       	2020-12-22 13:13:24.883791632 +0800 CST	deployed	kubeapps-5.0.0	2.0.1 
```

- Output of `kubectl version`:

```
root@lichenhao:~# kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:17:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:32:58Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by antgamdia** [comment]
Probably this issue will be solved once we release a new version with the changes of the PR https://github.com/kubeapps/kubeapps/pull/2218.

**Comment by DesistDaydream** [comment]
Supplementary note: In fact, this problem is because the data in postgres has not been deleted

```
assets=# select * from repos;
 id | namespace |        name        |                             checksum                             |                       last_update                       
----+-----------+--------------------+------------------------------------------------------------------+---------------------------------------------------------
 24 | test      | registory-bj-net-2 | b5c6c6107f390b709f6edbe5fd84edc294a334c0844004e49efa076d7662ee6e | 2020-12-24 07:43:52.141809969 +0000 UTC m=+1.694279946
 17 | kubeapps  | registory-bj-net   | b5c6c6107f390b709f6edbe5fd84edc294a334c0844004e49efa076d7662ee6e | 2020-12-24 07:50:08.182210414 +0000 UTC m=+3.376640020
 19 | default   | registory-bj-net   | b5c6c6107f390b709f6edbe5fd84edc294a334c0844004e49efa076d7662ee6e | 2020-12-24 07:55:41.293684049 +0000 UTC m=+2.128752863
 30 | test      | registory-bj-net   | b5c6c6107f390b709f6edbe5fd84edc294a334c0844004e49efa076d7662ee6e | 2020-12-24 08:12:54.745423569 +0000 UTC m=+2.837463365
  1 | kubeapps  | bitnami            | 58378d7ecd8430268c71e205085f89de054f22e7f876a08437c930130a69d671 | 2020-12-24 11:20:33.525261012 +0000 UTC m=+30.987448097
(5 rows)

```

**Comment by andresmgot** [comment]
Yes, after #2218 the cronjob and the data in postgres will be deleted so this should be fixed. What won't be deleted will be the first sync job created at the same time than the app repo but that shouldn't be a problem since the secret will be available at that time. Let me close this as a duplicate of #1832 but let us know if you still find any related issue after the next release.
